### ISSUE TITLE
Allow pre-release todo comments

### ISSUE THREAD

**Issue by thierry2015** [sindresorhus/eslint-plugin-unicorn#432]
As a developer, I want to be able to use todo comments with major release expiration while developing a pre-release version so I can progressively fix those todos without CI errors.

I am currently working on a 1.0.0-beta.1 version, and I want to deprecate a method, indicating that it will be removed in the official release (1.0.0). I want to have a `TODO` reminder that the method should be removed on `1.0.0`. However, the `unicorn/expiring-todo-comments` uses `semver.coerce`, which converts the `1.0.0-beta.1` to `1.0.0` when testing the `[>=1.0.0]` rule, which results in a linting error.

It would be a great improvement to expose a configuration to support pre-release todos.
Here would be, in my own opinion, some valid solutions for the plugin configuration.

```
"unicorn/expiring-todo-comments": [
        "error",
        {
                "allowPreRelease": true
        }
]

// or

"unicorn/expiring-todo-comments": [
        "error",
        {
                "allowPreRelease": [ // will use coercion for "rc" releases
                        "alpha"
                        "beta"
                ]
        }
]
```

## Pass
```
// package.json
{ "version": "1.0.0-beta.1" }

// my-file.js
// TODO [>=1.0.0] Something must be removed
```

## Fail
```
// package.json
{ "version": "1.0.0-beta.1" }

// my-file.js
// TODO [>=1.0.0-alpha.1]
```

**Comment by sindresorhus** [comment]
@lubien Thoughts?

**Comment by lubien** [comment]
First of all, thanks for pointing this issue @thierry2015 and for the clear explanation.

**tl;dr:** I think I found a solution that doesn't require any additional options.

----------

Explaining for future reference:

SO basically the point of `semver.coerce` was to handle not fully written versions such as `1` or `1.0`. Little did I know about how it handles pre-releases. 

As of [this](https://github.com/npm/node-semver#functions), apparently there is a flag `includePrerelease` you could put but it's not working as expected.

```
$ semver.coerce('1.0.0-beta.1', {includePrerelease: true})
=> SemVer {
  options: { includePrerelease: true },
  loose: false,
  raw: '1.0.0',
  major: 1,
  minor: 0,
  patch: 0,
  prerelease: [],
  build: [],
  version: '1.0.0' }
```

But this is fine. My solution was simple. Just don't coerce valid strings. Semver will understand "1.0.0-beta.1" even as string so I have no need to coerce this.

```js
function tryToCoerceVersion(version) {
	if (semver.valid(version)) {
		return version
	}

	try {
		return semver.coerce(version);
	} catch (_) {
		return false;
	}
}
```

[Working example](https://repl.it/@lubien/better-tryToCoerceVersion)

I'll work on a PR right now for this.

**Comment by thierry2015** [comment]
Great investigation job @lubien !
